x-pack/plugins/security/server/session_management/session_index.ts

```diff
@@ -318,11 +320,40 @@ export class SessionIndex {
     const sessionIndexTemplateName = `${this.options.kibanaIndexName}_security_session_index_template_${SESSION_INDEX_TEMPLATE_VERSION}`;
     return (this.indexInitialization = new Promise<void>(async (resolve, reject) => {
       try {
+        // Check if legacy index template exists, and remove it if it does.
```

Comment:
This seems prudent, but isn't the [legacy "Index template exists" API also deprecated](https://www.elastic.co/guide/en/elasticsearch/reference/7.16/indices-template-exists-v1.html)?

The [docs](https://www.elastic.co/guide/en/elasticsearch/reference/7.16/index-templates.html#index-templates) state this:
> The following conditions apply to index templates:
>
> * Composable templates take precedence over legacy templates. If no composable template matches a given index, a legacy template may still match and be applied.

So, perhaps we would be better off just ignoring the legacy index template. Unless you received other guidance from the Elasticsearch team?

Reply:
I almost 100% sure that it didn't generate a depreciation warning (unlike putTemplate), but it is a good question, let me double check that with the ES team.

Reply:
Really don't want to keep that garbage for the eternity, if possible :)